Fixed Power function of negative numbers

bobvandenberge opened the pull request:
Following issue: https://bz.apache.org/bugzilla/show_bug.cgi?id=62121

Commit:
Fixed Power function of negative numbers

asfgit commented:
Can one of the admins verify this patch?

pjfanning commented:
this change breaks a pre-existing unit test
unit tests can be run using `ant clean test`

  <testcase classname="org.apache.poi.ss.formula.eval.TestFormulasFromSpreadsheet" name="processFunctionRow[PowerEval]" time="0.006">
    <failure message="Function &apos;PowerEval&apos;: Formula: B13^0 @ 70:5: Expected 1.0 but was -1.0" type="junit.framework.AssertionFailedError">junit.framework.AssertionFailedError: Function &apos;PowerEval&apos;: Formula: B13^0 @ 70:5: Expected 1.0 but was -1.0
	at org.apache.poi.ss.formula.functions.AbstractNumericTestCase.assertEquals(AbstractNumericTestCase.java:64)
	at org.apache.poi.ss.formula.eval.TestFormulasFromSpreadsheet.processFunctionRow(TestFormulasFromSpreadsheet.java:207)
</failure>

Commit:
Fixed failing scenario for exponantiating a negative number to 0 and added test-cases to the FormulaEvalTestData.xls sheet

bobvandenberge commented:
@pjfanning thanks for the feedback. I updated the code as suggested and I added unit test cases for it. I also updated to excel sheet and included a test-scenario for the original problem.
